src/login/logind-core.c

```diff
@@ -407,10 +408,22 @@ int manager_get_user_by_pid(Manager *m, pid_t pid, User **ret) {
 int manager_get_idle_hint(Manager *m, dual_timestamp *t) {
         Session *s;
         bool idle_hint;
-        dual_timestamp ts = DUAL_TIMESTAMP_NULL;
+        dual_timestamp ts;
+        struct sysinfo info;
+        int r;
+        usec_t boot_usec;
 
         assert(m);
 
+        r = sysinfo(&info);
```

Comment:
urks, please, let's not use sysinfo(), it's an awful api.

is this really the uptime we want here? i mean, if we restart logind, then we'd return quite confusing values, i.e. the system might have idled/unidled before, but we'd suggest it wasn't.

I think it would make more sense to just take a timestamp when logind initializes and use that as base. any idleness info we have starts earliest at that time after all...